mail_tracking_mailgun/models/res_partner.py

```diff
@@ -31,28 +32,33 @@ def _email_bounced_set(self):
     def _mailgun_values(self):
         icp = self.env['ir.config_parameter']
         api_key = icp.get_param('mailgun.apikey')
-        api_url = icp.get_param('mailgun.api_url')
-        domain = icp.get_param('mailgun.mydomain')
+        if not api_key:
+            raise UserError(_('There is no Mailgun API!'))
+        api_url = icp.get_param(
+            'mailgun.api_url', 'https://api.mailgun.net/v3')
+        domain = icp.get_param('mail.catchall.domain')
         return api_key, api_url, domain
 
     @api.multi
     def check_email_validity(self):
-        """ Checks mailbox validity with Mailgun's API """
+        """Checks mailbox validity with Mailgun's API"""
         api_key, api_url, domain = self._mailgun_values()
         for partner in self:
             res = requests.get(
-                '%s/%s/address/validate --data-urlencode address=\'%s\'' % (
-                    api_url, domain, partner.email),
-                auth=("api", api_key))
-            if not res.mailbox_verification:
-                body = _('%s failed the mailbox verification. Please check it '
-                         'in order to avoid sending issues') % (partner.email)
-                partner.message_post(body=body)
+                "%s/address/validate" % api_url,
+                auth=("api", api_key), params={
+                    "address": partner.email,
+                    "mailbox_verification": True,
+                })
+            if not res['mailbox_verification']:
                 partner.email_bounced = True
+                raise ValidationError(
```

Comment:
And shouldn't this one be a `UserError`, given its purpose is to inform the user about some error? 😆